draft-ion.adoc

```diff
@@ -208,79 +200,176 @@ represented as follows:
+
+If a member name is not equal to the case-sensitive octet sequence `meta`, an Ion parser MUST NOT identify a member as a _Meta Object_.
+
+The member names within an identified _Meta Object_ MUST be unique; Ion parsers MUST either
+reject _Meta Object_ members with duplicate member names or use a JSON
+parser that returns only the lexically last duplicate member, as specified
 in {uri-ecmaScript-5-1-15-12}[Section 15.12 (The JSON Object)] of {uri-ecmaScript-5-1}[ECMAScript 5.1].
 
-The set of fields that a _Meta Object_ must contain to be considered valid
+The set of members that a _Meta Object_ must contain to be considered valid
 is resource-dependent.  In the absense of any resource-specific requirements,
-any field not understood by implementations MUST be ignored.
+any member not understood by implementations MUST be ignored.
+
+[[links]]
+== Links
+
+An Ion Link is an Ion Object where its `meta` member contains an {uri-IRIs}[IRI] member named `href`. An Ion Link is therefore JSON serialization of a {uri-rfc5988}[web link] to another resource.
+
+Ion parsers MUST identify any JSON structure matching this definition as an Ion Link.  Anything that does not match this defined structure MUST NOT be considered an Ion Link.
+
+A non-normative example of a minimal Ion link:
+
+.Example simplest Ion link:
+[source,json]
+----
+{ "meta": { "href": "https://api.example.com/corporations/acme" } }
+----
+
+=== Link Relation Types
+
+All Ion links have one or more <<links, link relation types>>.  A link's total set of link relation types is the set union of the link's implicit link relation type and the link's explicit link relation types.
+
+==== Implicit Link Relation Type
+
+All Ion links have exactly one implicit link relation type.  The implicit link relation type can be one of three possible values depending on where the link is located in the Ion content:
+
+1. Root Object
+2. Named Link
+3. Collection Resource `items` array element
+
+===== Root Object Implicit Link Relation Type
+
+If the Root Object is also found to be an Ion Link, it has an implicit link relation type of `self`.
+
+A non-normative example of a Root Object that is also an Ion Link:
+
+[source,json]
+----
+{
+  "meta": { "href": "https://api.example.com/users/1" },
+  "name": "Joe"
+}
+----
 
-=== Registered Metadata Field Names
+===== Named Link Implicit Link Relation Type
 
-The following field names are registered in the IANA Ion Metadata field registry
-defined in [TBD].  None of the fields defined below are intended to be mandatory
+If a JSON Object member value is an Ion Link, the Ion Link is implicitly assigned a link relation type that equals the member name.
+
+A non-normative example:
+
+[source,json]
+----
+{
+  "employer": { "meta": { "href": "https://api.example.com/corporations/acme" } }
+}
+----
+
+The above example shows an Ion Link member named `employer`.  The octet sequence `employer` is therefore assigned as the link's implicit link relation type.
```

Comment:
> A question comes to mind: do you consider the Collection Resource to be a best practice of REST?

I do.  Collection Resources are not strictly mandatory for REST interchange because as long as there are link transitions and no out-of-band info, you have REST.  But that's a bare minimum requirement.  Best practices usually go beyond the bare minimum to enable flexibility, power and ease of understanding on top of the bare minimum without sacrificing much (if anything).

I consider them a best practice for the following reasons:

1.  As we've said before, REST is all about link transitions and no out-of-band info.  To have no out-of-band info, you pretty much need metadata.  The only native collection concept in JSON is an array, and JSON arrays alone are incapable of natively supporting metadata (without attempting ugly or convoluted conventions).  Because JSON arrays can't do this, another concept needs to.  This is Ion's Collection Resource.

2.  Human beings are cognitively predisposed to grouping things of similar type - from mathematical set theory and computer science algorithms, to classifications of living things (kingdom, phylum, etc) to organization charts, etc, etc, etc.  Collections just 'makes sense' to pretty much everyone.  When it comes to data representations for exchange between services (Ion is a REST standard, but its focus is specifically 'data interchange' not just ad-hoc transitions) - and those services are written and consumed by engineers writing software, a proper Collection concept just 'makes sense' to me.

3.  Segmentation.  Without having a mandate that a Collection Resource has exactly 1 array (default named `items`), multiple arrays could be introduced in a single JSON object and parsing metadata for each respective array could be confusing/complex (because it's all jumbled/aggregated in the resource's `meta` field).  While this is possible, it's ugly, and this kind of overloading will inherently cause what I call 'cognitive friction' (i.e. while we can make rules that a computer can follow, it's just a huge pain for a human - not nice).

Number 1 is probably most important from a specification perspective because it closes a glaring hole in JSON for REST purposes.  

Numbers 2 and 3 is what makes Ion Ion - human friendliness and ease of use/understanding is (almost) an equal goal of the spec.  In other words, my vision for Ion is not only to come up with a spec that machine clients can utilize (and abstract away from humans), but one that humans can read and write natively - without clients or tools - because it is easy enough to do so and it just 'feels' good.  Yes, that is totally subjective, but I think we're already pretty close. :)

On this last point, I feel that HTML is similar - it can be read/written by humans without tools of any kind, yet tools (browsers, IDEs, server rendering engines) work wonderfully with it.  I feel this is the 'sweet spot' for spec adoption and ubiquity.

HTH!